Software/src/inverter/BYD-MODBUS.cpp

```diff
@@ -56,12 +56,12 @@ void handle_update_data_modbusp201_byd() {
 void handle_update_data_modbusp301_byd() {
   // Store the data into the array
   static uint16_t battery_data[24];
-  if (battery_current > 0) {         //Positive value = Charging
-    bms_char_dis_status = 2;         //Charging
-  } else if (battery_current < 0) {  //Negative value = Discharging
-    bms_char_dis_status = 1;         //Discharging
-  } else {                           //battery_current == 0
-    bms_char_dis_status = 0;         //idle
+  if (battery_current == 0) {
+    bms_char_dis_status = STANDBY;
+  } else if (battery_current > 32768) {  //Negative value = Discharging
```

Comment:
32767 is the maximum positive representation, so
```suggestion
  } else if (battery_current > 32767) {  //Negative value = Discharging
```
would be preferable